Microsoft.SystemForCrossDomainIdentityManagement/Service/RequestExtensions.cs

```diff
@@ -62,5 +65,214 @@ public static bool TryGetRequestIdentifier(this HttpRequestMessage request, out
+                dynamic operationDataJson = JsonConvert.DeserializeObject(context.Operation.Data.ToString());
+                IReadOnlyCollection<PatchOperation2Combined> patchOperations = operationDataJson.Operations.ToObject<List<PatchOperation2Combined>>();
+                PatchRequest2 patchRequest = new PatchRequest2(patchOperations);
+
+                foreach (IBulkCreationOperationContext creation in creations)
+                {
+                    if (null == creation.Operation)
+                    {
+                        throw new InvalidOperationException(
+                            SystemForCrossDomainIdentityManagementServiceResources.ExceptionInvalidOperation);
+                    }
+
+                    if (string.IsNullOrWhiteSpace(creation.Operation.Identifier))
+                    {
+                        throw new InvalidOperationException(
+                            SystemForCrossDomainIdentityManagementServiceResources.ExceptionInvalidOperation);
+                    }
+
+                    if (patchRequest.References(creation.Operation.Identifier))
+                    {
+                        creation.AddDependent(context);
+                        context.AddDependency(creation);
+                    }
+                }
+            }
+            catch
+            {
+                throw new HttpResponseException(HttpStatusCode.BadRequest);
+            }
+                   
+        }
+        
+
+        private static void Enlist(
+            this IRequest<BulkRequest2> request,
+            BulkRequestOperation operation,
+            List<IBulkOperationContext> operations,
+            List<IBulkCreationOperationContext> creations,
+            List<IBulkUpdateOperationContext> updates)
+        {
+            if (null == operation)
+            {
+                throw new ArgumentNullException(nameof(operation));
+            }
+
+            if (null == operations)
+            {
+                throw new ArgumentNullException(nameof(operations));
+            }
+
+            if (null == creations)
+            {
+                throw new ArgumentNullException(nameof(creations));
+            }
+
+            if (null == updates)
+            {
+                throw new ArgumentNullException(nameof(updates));
+            }
+
+            if (null == operation.Method)
+            {
+                throw new ArgumentException(SystemForCrossDomainIdentityManagementServiceResources.ExceptionInvalidOperation);
+            }
+
+            if (HttpMethod.Post == operation.Method)
+            {
+                IBulkCreationOperationContext context = new BulkCreationOperationContext(request, operation);
+                context.Relate(updates);
+                var firstDependent =
+                    operations
+                    .Select(
+                        (IBulkOperationContext item, int index) =>
+                            new { Item = item, Index = index })
```

Comment:
One never needs to use anonymous types anymore. One can use deconstructed tuples instead now: 

Select(
    (IBulkOperationContext item, int index) => (item, index))
.Where(
    ((X Item, int Index) item) =>